release-notes/7.0/preview/api-diff/preview1/Microsoft.NETCore.App/7.0-preview1_System.Numerics.md

```diff
@@ -0,0 +1,67 @@
+# System.Numerics
+
+``` diff
+ namespace System.Numerics {
+     public static class BitOperations {
++        public static bool IsPow2(IntPtr value);
++        public static bool IsPow2(UIntPtr value);
++        public static int LeadingZeroCount(UIntPtr value);
++        public static int Log2(UIntPtr value);
++        public static int PopCount(UIntPtr value);
++        public static UIntPtr RotateLeft(UIntPtr value, int offset);
++        public static UIntPtr RotateRight(UIntPtr value, int offset);
++        public static UIntPtr RoundUpToPowerOf2(UIntPtr value);
++        public static int TrailingZeroCount(IntPtr value);
++        public static int TrailingZeroCount(UIntPtr value);
+     }
+     public struct Matrix3x2 : IEquatable<Matrix3x2> {
++        public float this[int row, int column] { get; set; }
+     }
+     public struct Matrix4x4 : IEquatable<Matrix4x4> {
++        public float this[int row, int column] { get; set; }
+     }
+     public struct Quaternion : IEquatable<Quaternion> {
++        public float this[int index] { get; set; }
++        public static Quaternion Zero { get; }
+     }
+     public static class Vector {
++        public static Vector<Byte> ShiftLeft(Vector<Byte> value, int shiftCount);
```

Comment:
A "nit" on the tooling is that generics show as the .NET types (`Byte`) rather than the language keywords (`byte`). This is "different" from a lot of the rest of the diffs here.

Reply:
This issue would go into arcade. The tool that generates this is Microsoft.DotNet.AsmDiff

I would suggest to rewrite at some point that tool which uses CCI, on top of the new API Compat object model wrote in the SDK on top of Roslyn.

Reply:
Feel free to open a new issue, or you can mention this related problem in the issue I opened to track `nint` and `nuint`: https://github.com/dotnet/arcade/issues/8484